### PR TITLE
Update au-core-obs-01 to support DAR extension and change the length of dateTime to >= 10 (FHIR-46407)

### PR DESCRIPTION
**Pull request by dbojicic**
This PR provides fix for [FHIR-46407](https://jira.hl7.org/browse/FHIR-46407).

### CODE REVIEW

File: input/resources/au-core-diagnosticresult-path.xml [hunk 1]
```diff
@@ -141,8 +141,8 @@
         <key value="au-core-obs-01"/>
         <severity value="error"/>
         <human value="Date shall be at least to day"/>
```
Comment (JayMurdoch) [inline input/resources/au-core-diagnosticresult-path.xml:143]: Suggest updating to reflect addition of DAR extension eg "Date shall be at least to day or, if not available, the Data Absent Reason extension shall be present"

File: input/resources/au-core-diagnosticresult.xml [hunk 2]
```diff
@@ -138,8 +138,8 @@
         <key value="au-core-obs-01"/>
         <severity value="error"/>
         <human value="Date shall be at least to day"/>
```
Comment (JayMurdoch) [inline input/resources/au-core-diagnosticresult.xml:140]: Suggest updating to reflect addition of DAR extension eg "Date shall be at least to day or, if not available, the Data Absent Reason extension shall be present"

**Review by JayMurdoch — CHANGES_REQUESTED** [review]
Suggest change log entry given the change to from 8 to 10 characters.

See inline comments.

File: input/resources/au-core-diagnosticresult.xml (hunk 2 above)
Comment (dbojicic) [reply input/resources/au-core-diagnosticresult.xml]: Done

**Review by brettesler-ext — APPROVED** [review]

**Pull request merged**